k-distribution/src/main/scripts/bin/kx

```diff
@@ -0,0 +1,438 @@
+#!/bin/bash
+set -e
```

Comment:
I think we should `set -eo pipefail` at least, so any commands added in pipes also fail the script if they fail (instead of just the last command in the pipe).

We probably also should do `set -euo pipefail`, so that we fail on unitialized variables (and so changes to variables here must provide defaults), but that may be trickier.